### PR TITLE
fix issue #110 kotlin-dsl

### LINKED ISSUE #110: [Assigned] Kotlinize build.gradle
## Overview (Required)
- We can use build.gradle.kts instead of build.gradle
- I think Kotlin is great. So we should use it 👍 

## Links
- https://kotlinlang.org/docs/reference/using-gradle.html

### PR DESCRIPTION
**Pull request by shikajiro**
## Issue
- close #110

## Overview (Required)
- build.gradle files change to build.gradle.kts
- create buildSrc dir.  Versions.kt file is management library versions.
ex:
```
object Versions{
  const val compileSdk = 27
}

android{
  compileSdkVersion(Versions.compileSdk)
}
```
I made a kotlin file without using `ext`.
Because code completion can be used.

- Depends.kt file is management dependencies libraries.

ex:
```
object Depends {
    object Kotlin {
        val stdlib = "org.jetbrains.kotlin:kotlin-stdlib-jre7:${Versions.kotlin}"
...

implementation(Depends.Kotlin.stdlib)
```

- settings.gradle can not change to kts. It causes some compile error 😣

- I checked build & run. But I can not check deploy to Google Play.

## Links
- https://github.com/gradle/kotlin-dsl
### respect
- https://qiita.com/chibatching/items/d63c0c9afc8e1e6bc9de
- https://qiita.com/chibatching/items/3b40aea222b2a2f29d4b
- https://github.com/chibatching/KtsSample
- https://github.com/jaredsburrows/android-gif-example
- https://github.com/justjanne/QuasselDroid-ng

## Screenshot
no file

### CODE REVIEW

**Comment by takahirom** [comment]
🆒

File: buildSrc/gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -0,0 +1,5 @@
+distributionBase=GRADLE_USER_HOME
+distributionPath=wrapper/dists
+zipStoreBase=GRADLE_USER_HOME
+zipStorePath=wrapper/dists
+distributionUrl=https\://services.gradle.org/distributions/gradle-4.3.1-all.zip
```
Comment (takahirom) [inline buildSrc/gradle/wrapper/gradle-wrapper.properties]: Can we use 4.5? 🤔

Comment (shikajiro) [reply buildSrc/gradle/wrapper/gradle-wrapper.properties]: oh! It was old!

Comment (shikajiro) [reply buildSrc/gradle/wrapper/gradle-wrapper.properties]: nnn.....? I feel I do not need this. I'll fix it.

File: app/build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+
+    })
+    signingConfigs {
+        getByName("debug") {
+            storeFile = rootProject.file("debug.keystore")
+            storePassword = "android"
+            keyAlias = "androiddebugkey"
+            keyPassword = "android"
+        }
+        create("release") {
```
Comment (takahirom) [inline app/build.gradle.kts]: I have a question 🙋 
How do you use "getByName" and "create" differently?
(どうやって使い分けています？？ 🤔 )

Comment (shikajiro) [reply app/build.gradle.kts]: I also use `getByName("release")` at first.
but i got an error.
`Error:SigningConfig with name 'release' not found.`

Perhaps at this point it seems that 'release' does not exit yet.

（最初はgetByNameで試したのですが、「releaseはまだないよ」ってエラーになりました。恐らく最初の時点では release が存在してないと思われます）

Comment (takahirom) [reply app/build.gradle.kts]: Thanks! I understood! If we got the error when using getByName, we should change getByName! 😄 

エラー出たらcreateに変える感じなんですね！

**Comment by takahirom** [comment]
LGTM 👍 
I want to try that 😄

**Pull request merged**